src/fides/ctl/core/export.py

```diff
@@ -174,45 +176,67 @@ def generate_system_records(
-                )
-                for category in data_categories
-                for subject in data_subjects
-                for dataset_reference in dataset_references
+        if system.privacy_declarations:
+            for declaration in system.privacy_declarations:
+                data_use = formatted_data_uses[declaration.data_use]
+                data_categories = declaration.data_categories or []
+                data_subjects = [
+                    formatted_data_subjects[data_subject_fides_key]
+                    for data_subject_fides_key in declaration.data_subjects
+                ]
+                dataset_references = declaration.dataset_references or [
+                    EMPTY_COLUMN_PLACEHOLDER
+                ]
+                cartesian_product_of_declaration = [
+                    (
+                        system.fides_key,
+                        system.name,
+                        system.description,
+                        system.data_responsibility_title,
+                        system.administrating_department,
+                        third_country_list,
+                        system_dependencies,
+                        declaration.name,
+                        category,
+                        data_use["name"],
+                        data_use["legal_basis"],
+                        data_use["special_category"],
+                        data_use["recipients"],
+                        data_use["legitimate_interest"],
+                        data_use["legitimate_interest_impact_assessment"],
+                        subject["name"],
+                        subject["rights_available"],
+                        subject["automated_decisions_or_profiling"],
+                        declaration.data_qualifier,
+                        data_protection_impact_assessment["is_required"],
+                        data_protection_impact_assessment["progress"],
+                        data_protection_impact_assessment["link"],
+                        dataset_reference,
+                    )
+                    for category in data_categories
+                    for subject in data_subjects
+                    for dataset_reference in dataset_references
+                ]
+                output_list += cartesian_product_of_declaration
+        else:
+            system_row = [
+                system.fides_key,
+                system.name,
+                system.description,
+                system.data_responsibility_title,
+                system.administrating_department,
+                third_country_list,
+                system_dependencies,
+                data_protection_impact_assessment["is_required"],
+                data_protection_impact_assessment["progress"],
+                data_protection_impact_assessment["link"],
             ]
-            output_list += cartesian_product_of_declaration
+            length_to_na = 12
+            start_position = 6
```

Comment:
it took me a bit to understand what this was doing. I wonder if naming these variables something more like 

```
num_privacy_declaration_fields = 12
privacy_declaration_start_index = 6
```

would be clearer? then again, I haven't really worked on this set of code before, so it might just be perfectly clear if I had more context :)

Reply:
I like your naming a lot better, thanks! I almost even just left it explicitly/static as `"N/A"` repeatedly just to make it clear but very happy to go in the direction you suggest 👍🏽